be/src/pipeline/exec/exchange_sink_operator.cpp

```diff
@@ -155,12 +155,17 @@ Status ExchangeSinkLocalState::open(RuntimeState* state) {
         size_t dep_id = 0;
         for (auto* channel : channels) {
             if (channel->is_local()) {
-                _local_channels_dependency.push_back(channel->get_local_channel_dependency());
```

Comment:
why should we extract this dependency dynamically?
The channel is inited during prepare.
We should extract these dependency during open？
I think we should extract it during prepare stage and then we could avoid such problems.